src/interceptors/xhrInterceptor.js

```diff
@@ -108,13 +109,17 @@ class FakeXMLHttpRequest {
             body: data,
             headers: this._requestHeaders
           });
-          const response = KakapoResponse.wrap(handler(request, db));
-
-          if (delay) {
-            setTimeout(() => this._handleResponse(response), delay);
-          } else {
-            this._handleResponse(response);
-          }
+          // Wrapping handler into a promise to add promise support for free
+          const responsePromise = Promise.resolve(handler(request, db));
+          
+          responsePromise.then((result) => {
```

Comment:
I tried to use **await** instead but the example was failing due babel runtime generator missing :/